bgls/near_clifford_solver.py

```diff
@@ -0,0 +1,93 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation of additional Clifford circuit handling not yet used in
+simulator."""
+
+from typing import List, Tuple
+
+import numpy as np
+
+import cirq
+
+
+def is_clifford_gate(op: cirq.Operation) -> bool:
+    """
+    More strictly than cirq.has_stabilizer_effect(op), yields whether the
+    gate is in the Clifford group,
+    only allowing explicitly H, S, CNOT, and Measurement gates.
```

Comment:
Why not allow Clifford gates outside of H, S, CNOT, M?

Reply:
agreed, I removed this, I just hadn't been sure if there were corner cases that has_stabilizer_effect allowed that we didn't want, but I think that should be adequate

Reply:
Yes it should be safe - LGTM